src/ingestion-server/kafka-s3-connector/custom-resource/kafka-s3-sink-connector/index.ts

```diff
@@ -607,3 +612,13 @@ async function download(url: string, outPath: string) {
     });
   });
 }
+
+async function updateConnectorTags(context: Context, connectorArn?: string) {
+  const tags = await getFunctionTags(context);
+  logger.info('funcTags', { tags });
+  const command = new TagResourceCommand({
```

Comment:
Did you test upgrading case with different tags?

Reply:
Good point, 
Refactored code to handle there are some tags need to be deleted and some are added.